_posts/2018-11-27-pool-features.md

```diff
@@ -0,0 +1,149 @@
+---
+title: Pool features
+author: jmmichal
+layout: post
+identifier: pool_features
+---
+
+# Introduction
+
+From the beginning of persistent memory pools existence pools have an internal state called features. It was not very useful for the end user to know what features are enabled in his pools so it was an implementation detail. The more vigilant user may notice pool features was listed in the [pmempool info][pmempool-info.1] output but not much more.
+
+Starting from [release 1.5][release-1.5] set of new features have been introduced. It is **crucial for data safety** to know if e.g. [Reliability, Availability and Serviceability \(RAS\)][RAS] features are enabled in the pool or not. To meet the need bunch of new possibilities are now available:
+
+* `pmempool` tool [commands for managing features][pmempool-feature.1]
+* *libpmempool* [API for managing features][pmempool-feature.3]
+* CTL namespace for [setting a `SHUTDOWN_STATE` initial value](#set-shutdown_state-initial-value)
+
+# The features taxonomy
+
+Any feature can be disabled or enabled in the pool and supported or not supported in the used version of the PMDK library. All features are divided into three groups:
```

Comment:
Any one feature can be..., and the support for those features might vary depending on the library version.